app/src/main/java/com/sun/chat_04/data/remote/FriendRemoteDataSource.kt

```diff
@@ -0,0 +1,41 @@
+package com.sun.chat_04.data.remote
+
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import com.sun.chat_04.data.model.Friend
+import com.sun.chat_04.data.repositories.FriendDataSource
+import com.sun.chat_04.ui.signup.RemoteCallback
+import com.sun.chat_04.util.Constants
+
+class FriendRemoteDataSource(
+    private val auth: FirebaseAuth,
+    private val database: FirebaseDatabase
+) : FriendDataSource.Remote {
+
+    override fun getFriends(callback: RemoteCallback<ArrayList<Friend>>) {
```

Comment:
```suggestion
    override fun getFriends(userId: String, callback: RemoteCallback<ArrayList<Friend>>) {
```
em nên viết ntn 
tăng tính tái sử dụng của code,
giảm sự phụ thuộc của FriendRemoteDataSource với FirebaseAuth

Reply:
vâng em truyền userId thay vì firebase auth vào datasource rồi ạ

Reply:
em nên truyền vào function ntn thì hợp lý hơn
user id k nhất thiết phải khai báo là instance variable